lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -3785,6 +3771,124 @@ ParseResult FIRStmtParser::parseInstance() {
+  if (auto isExpr = parseExpWithLeadingKeyword(startTok))
+    return *isExpr;
+
+  if (requireFeature(nextFIRVersion, "options"))
+    return failure();
+
+  LocWithInfo info(getToken().getLoc(), this);
+
+  StringRef id;
+  StringRef defaultModuleName;
+  StringRef optionGroupName;
+  if (parseId(id, "expected instance name") ||
+      parseToken(FIRToken::kw_of, "expected 'of' in instance") ||
+      parseId(defaultModuleName, "expected module name") ||
+      parseId(optionGroupName, "expection option group name") ||
+      info.parseOptionalInfo())
+    return failure();
+
+  locationProcessor.setLoc(startTok.getLoc());
+
+  // Look up the default module referenced by the instance choice.
+  // The port lists of all the other referenced modules must match this one.
+  auto defaultModule = getReferencedModule(loc, defaultModuleName);
+  if (!defaultModule)
+    return failure();
+
+  SmallVector<PortInfo> modulePorts = defaultModule.getPorts();
+
+  // Find the option group.
+  auto circuit =
+      builder.getBlock()->getParentOp()->getParentOfType<CircuitOp>();
+  auto optionGroup =
+      dyn_cast_or_null<OptionOp>(circuit.lookupSymbol(optionGroupName));
+  if (!optionGroup)
+    return emitError(loc,
+                     "use of undefined option group '" + optionGroupName + "'");
+
+  auto baseIndent = getIndentation();
+  SmallVector<std::pair<OptionCaseOp, FModuleLike>> caseModules;
+  while (getIndentation() == baseIndent) {
+    StringRef caseId;
+    StringRef caseModuleName;
+    if (parseId(caseId, "expection a case identifier") ||
+        parseToken(FIRToken::equal_greater,
+                   "expected '=> in instance choice definition") ||
+        parseId(caseModuleName, "expected module name"))
+      return failure();
+
+    auto caseModule = getReferencedModule(loc, caseModuleName);
+    if (!caseModule)
+      return failure();
+
+    for (const auto &[defaultPort, casePort] :
+         llvm::zip(modulePorts, caseModule.getPorts())) {
+      if (defaultPort.name != casePort.name)
+        return emitError(loc, "instance case module port '")
+               << casePort.name.getValue()
+               << "' does not match the default module port '"
+               << defaultPort.name.getValue() << "'";
+      if (defaultPort.type != casePort.type)
+        return emitError(loc, "instance case port '")
+               << casePort.name.getValue()
+               << "' type does not match the default module port";
+    }
+
+    auto optionCase =
+        dyn_cast_or_null<OptionCaseOp>(optionGroup.lookupSymbol(caseId));
+    if (!optionCase)
+      return emitError(loc, "use of undefined option case '" + caseId + "'");
+    caseModules.emplace_back(optionCase, caseModule);
+  }
+
+  auto annotations = getConstants().emptyArrayAttr;
+  SmallVector<Attribute, 4> portAnnotations(modulePorts.size(), annotations);
+
+  // Create an instance choice op.
+  StringAttr sym = {};
```

Comment:
Nit: StringAttr probably doesn't need the empty initializer list? Is the following fine:

```
StringAttr sym;
```